src/NLog/Layouts/CsvLayout.cs

```diff
@@ -301,5 +301,24 @@ protected override void RenderFormattedMessage(LogEventInfo logEvent, StringBuil
                 _parent.RenderHeader(target);
             }
         }
+
+        /// <summary>
+        /// Generate description of CSV Layout
+        /// </summary>
+        /// <returns>CSV Layout String Description</returns>
+        public override string ToString()
+        {
+            if (Columns?.Count > 0)
+            {
+                var columnSelection = System.Linq.Enumerable.Select(Columns, c => c.Name);
+#if NET3_5
+                var columnNames = System.Linq.Enumerable.ToArray(columnSelection);
+#else
+                var columnNames = columnSelection;
+#endif
```

Comment:
could you please move this to a CollectionHelpers? (or LinqHelpers)

Reply:
Have sacrificed some memory by always doing ToArray before calling string.Join